### PR TITLE
[MNOE-574] - Improve error management

### PR DESCRIPTION
**Pull request by x4d3**
I thought we could improve the error management in MNOE by using a custom error catched on the controller level.

@BrunoChauvet  @alachaum  what do you think ?

This is waiting on #521 to be merged

It is also waiting on https://github.com/chingor13/json_api_client/pull/275

### CODE REVIEW

**Comment by BrunoChauvet** [comment]
I like this way of handling errors, way more concise and we end-up with leaner controllers :+1:

**Comment by x4d3** [comment]
@ouranos  PR ready for review

File: core/lib/mno_enterprise/concerns/models/team.rb [hunk 1]
```diff
@@ -24,29 +24,6 @@ module ClassMethods
   #==================================================================
   # Instance methods
   #==================================================================
-  # Add a user to the team
```
[the remark this answers is not in the document]
Comment (x4d3) [reply core/lib/mno_enterprise/concerns/models/team.rb]: Yes it is, they are actually not called in the current code.

**Comment by x4d3** [comment]
@ouranos  I've addressed your comment. Regarding the error management, it's indeed changing, but I don't think we were using the result array or hash properly on the frontend.

**Comment by ouranos** [comment]
It's not used properly at the moment but @alexnoox was looking into it.
The previous format was structured which would make it easy to parse and format in the frontend.
This new one, not so much.

**Comment by x4d3** [comment]
@ouranos  I've addressed your concerns. The json returned will be the same and follow that format
```{"name":["is too short (minimum is 2 characters)"]}```

**Comment by alexnoox** [comment]
Good work, it is much cleaner!

For the error handler in the frontends, we should do it in an interceptor checking if the response is an error. That would allow us to remove all the error management in the services. The errors will need to be re-thrower from the handler to catch them in the controllers if we need to display a toastr or errors on a form.

**Comment by x4d3** [comment]
@ouranos I've fixed the specs.

**Pull request merged**